### PR TITLE
[RFC] [DO NOT MERGE] smart_amp: fix logic on pause/stop

### PR DESCRIPTION
**Pull request by plbossart**
Add the logic already present in mixer but limited to the feedback
source. We have to keep track of the feedback source state, in case
it's paused/stopped.

This is an untested patch just to show what's missing at the moment
compared to the mixer.

FIXME:
1. I don't know if the logic is correct, it's different from what
the mux implements (https://github.com/thesofproject/sof/issues/3836)
2. Not sure if we need to add zeroes on the regular source as done by
the mixer.

BugLink: https://github.com/thesofproject/sof/issues/3837
Signed-off-by: Pierre-Louis Bossart <pierre-louis.bossart@linux.intel.com>

### CODE REVIEW

File: src/audio/smart_amp/smart_amp.c [hunk 1]
```diff
@@ -517,6 +528,20 @@ static int smart_amp_trigger(struct comp_dev *dev, int cmd)
...
 
 	ret = comp_set_state(dev, cmd);
+	if (ret < 0)
+		return ret;
+
+	if (ret == COMP_STATUS_STATE_ALREADY_SET)
+		return PPL_STATUS_PATH_STOP;
+
+	/* don't stop smart_amp on pause or if the feedback source is active/paused */
+	if (cmd == COMP_TRIGGER_PAUSE ||
```
Comment (cujomalainey) [inline src/audio/smart_amp/smart_amp.c:538]: this isn't fun to read, it looks like you can break this if up into a couple different checks with the same exit, please do.

Comment (plbossart) [reply src/audio/smart_amp/smart_amp.c:538]: this is copy/pasted from the mixer, and I filed a separate issue to figure out what the behavior is supposed to be. 

see https://github.com/thesofproject/sof/issues/3836, this whole thing is a real mess.

Comment (cujomalainey) [reply src/audio/smart_amp/smart_amp.c:538]: sounds like we are in a fun state regarding multi-input components

Comment (plbossart) [reply src/audio/smart_amp/smart_amp.c:538]: yep. Fun as in 'confused'.

I honestly don't know why this smart-amp component needed to be special cased with a source and a 'feedback' source. They are both sources.
when we have stuff like AEC, are we also going to special case?

It seems like we are not making use of proper abstractions or 'object hiearchy'. Changes on the sources should be dealt with in the same manner, be it the second source of a mixer or a feedback. this feels like we are still going from a linear pipeline to a graph and haven't fully conceptualized what's missing.

Comment (cujomalainey) [reply src/audio/smart_amp/smart_amp.c:538]: Do we have the ability to distinguish sources by anything other than index? if yes then I agree we should collapse them

**Pull request closed without merging**